### PR TITLE
Better error message.

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: theano/tensor/raw_random.py [hunk 1]
```diff
@@ -379,10 +379,14 @@ def _infer_ndim_bcast(ndim, shape, *args):
...
-            ndim = tensor.get_vector_length(v_shape)
-        bcast = [False] * ndim
 
+    if v_shape.ndim != 1:
+        raise TypeError("shape must be a vector or list of scalar, got '%s'" %
+                        v_shape)
+
+    if ndim is None:
+        ndim = tensor.get_vector_length(v_shape)
+        bcast = [False] * ndim
```
Comment (lamblin) [inline theano/tensor/raw_random.py]: This line should be outside of the `if` block, otherwise `bcast` does not get defined.

Comment (nouiz) [inline theano/tensor/raw_random.py]: bcast have already been defined if this if is False. The logic isn't clear, but it prevent duplicate code. In the other if/ifelse then the else clause that I changed, bast is defined.

Comment (lamblin) [inline theano/tensor/raw_random.py]: Well, some tests are failing because `bcast` is used before being assigned.

**Comment by nouiz** [comment]
@lamblin do you have the time to review this?

**Pull request merged**